### PR TITLE
clientv3/integration: test lease keepalive

### PR DESCRIPTION
**Pull request by xiang90**
/cc @heyitsanthony @gyuho

### CODE REVIEW

**Comment by gyuho** [comment]
LGTM.

File: clientv3/integration/lease_test.go [hunk 1]
```diff
@@ -73,3 +74,95 @@ func TestLeaseRevoke(t *testing.T) {
...
+
+	rc, kerr := lapi.KeepAlive(context.Background(), lease.LeaseID(resp.ID))
+	if kerr != nil {
+		t.Errorf("failed to keepalive lease %v", kerr)
+	}
+
+	kresp := <-rc
+	if kresp.ID != resp.ID {
+		t.Errorf("ID = %x, want %x", kresp.ID, resp.ID)
+	}
```
Comment (heyitsanthony) [inline clientv3/integration/lease_test.go]: test the channel closes on Close()?

Comment (xiang90) [inline clientv3/integration/lease_test.go]: sure. will do.

**Comment by heyitsanthony** [comment]
can this test the ctx cancel logic too?

**Comment by xiang90** [comment]
@heyitsanthony I wanted. However the keepAlive ctx is unimplemented right now. I plan to implement and then test it as a whole.

**Comment by heyitsanthony** [comment]
ok, fair enough

**Comment by xiang90** [comment]
@heyitsanthony fixed a bunch of small issues. PTAL.

File: clientv3/client.go [hunk 2]
```diff
@@ -182,6 +187,10 @@ func (c *Client) retryConnection(oldConn *grpc.ClientConn, err error) (*grpc.Cli
 // connection is established.
 func dialEndpointList(c *Client) (*grpc.ClientConn, error) {
 	var err error
+	if len(c.Endpoints()) == 0 {
```
Comment (heyitsanthony) [inline clientv3/client.go]: would it be better to put this in `New` instead of checking in the retrydialer every redial?

Comment (xiang90) [inline clientv3/client.go]: ok will do.

**Comment by heyitsanthony** [comment]
lgtm after fixing nits

**Comment by xiang90** [comment]
close in favor of https://github.com/coreos/etcd/pull/4382

**Pull request closed without merging**